### ISSUE TITLE
MongoDB  Transactions  requires replicas

### ISSUE THREAD

**Issue by feritzcan2** [microsoft/aspire#5238]
Hello,  we are using MongoDB and recently we have integrated with aspire.  
However, running MongoDB transactions requires replicas.  As I checked I couldn't find any solution right now.

Can you assist me if its possible please?

**Comment by jakoss** [comment]
We are also interested in that. It became more important after MongoDB EF Provider 8.1 enabled transactions by default and for now we have to explicitly disable those for local work

**Comment by twsouthwick** [comment]
I opened a PR that enables with the extension method:

```diff
namespace Aspire.Hosting;

public static class MongoDBBuilderExtensions
{
+  public static IResourceBuilder<MongoDBServerResource> WithReplicaSet(this IResourceBuilder<MongoDBServerResource> builder, string? replicaSet = null);
}
```

**Comment by feritzcan2** [comment]
> I opened a PR that enables with the extension method:
> 
> 
> 
> ```diff
> 
> namespace Aspire.Hosting;
> 
> 
> 
> public static class MongoDBBuilderExtensions
> 
> {
> 
> +  public static IResourceBuilder<MongoDBServerResource> WithReplicaSet(this IResourceBuilder<MongoDBServerResource> builder, string? replicaSet = null);
> 
> }
> 
> ```

How can i track status?

**Comment by jakoss** [comment]
> How can i track status?

Here: https://github.com/dotnet/aspire/pull/5712

It's also linked in the right menu of this issue

**Comment by feritzcan2** [comment]
@jakoss  is it gonna be published any soon? Pr was open since september :(

**Comment by davidfowl** [comment]
No, this PR needs to be replat on top of a new capability that isn't yet implemented. Being able to run commands in the container, on   start. Either that or we change the entrypoint of the build in container to be a shell script that does this.

**Comment by davidfowl** [comment]
Actually, I'll follow up with @twsouthwick to see if we can pull something off for 9.x, there may be options here.

**Comment by alexjamesbrown** [comment]
The previous workaround from @vzam here (https://github.com/dotnet/aspire/issues/6811) no longer works with Aspire 9.1.0

I've got a demo of it working in Aspire 9.0.0
https://github.com/alexjamesbrown/aspire-mongodb-replicaset

Upgrading to 9.1.0  - and it no longer works
https://github.com/alexjamesbrown/aspire-mongodb-replicaset/tree/9.1.0
Mainly because `builder.Eventing.Subscribe<ConnectionStringAvailableEvent>` no longer fires

https://github.com/alexjamesbrown/aspire-mongodb-replicaset/blob/9.1.0/AspireMongo.AppHost/MongoReplicaSetBuilderExtensions.cs#L40

**Comment by vzam** [comment]
@alexjamesbrown I don't know how the health check worked before, but we are using the database there, while the health check is only specified on the server. To fix this, go to `MongoDbReplicaSetBuilderExtensions.AddMongoReplicaSet`, rename `mongoDbServerResource` to just `mongoDbResource` and introduce this variable
```csharp
var mongoDbServerResource = mongoDbResource switch
        {
            MongoDBDatabaseResource dbResource => dbResource.Parent,
            MongoDBServerResource serverResource => serverResource,
            _ => mongoDbResource
        };
```
replace the two occurrences of `mongoDbResource` with `mongoDbServerResource` where the health checks are being removed.

**Comment by max2020204** [comment]
How soon will be method for configure replicas?

**Comment by twsouthwick** [comment]
I started looking at this again yesterday to revamp the PR I had submitted with some input from @davidfowl and @eerhardt. It's still kinda tricky with the available tooling but hope to have an updated approach later today or tomorrow.

**Comment by max2020204** [comment]
I saw you make pull request to main branch, when it will be approved?

**Comment by bkershaw** [comment]
If anyone runs into trouble getting a MongoDB replica set running inside Aspire, I found this repo super helpful:
👉 [AspireMongoReplicaSet](https://github.com/joewashington75/AspireMongoReplicaSet)

It really saved me some time while we’re all waiting for official support.

Hope it helps someone else too! 😊

**Comment by rbaudrillard** [comment]
I am starting a new project and i'm also interested in this feature. 
I would like to avoid setuping a container image my self, as for now all I needed was supported by aspire directly

Great product by the way :)

Can we just get a quick status on this one ?

**Comment by fedarovich** [comment]
As AspireMongoReplicaSet project mentioned above did not work for me well enough in my scenario (persistent instance accessible from both localhost and container network) on Aspire 13.x, I've created another implementation:

https://github.com/fedarovich/aspire-hosting-mongodb-replicaset

It uses Mongo DB [horizons](https://www.percona.com/blog/using-replicasethorizons-in-mongodb/) feature, and thus allows connections from both localhost and Aspire container network, works correctly after restart and supports both single-instance replica sets and multiple instance replica sets.

**Comment by artiomchi** [comment]
Taking another stab at this with a simplified approach that doesn't require building a new container image with a dockerfile. See PR above

**Comment by artiomchi** [comment]
Unfortunately the PR I created isn't getting any eyes on it in the last month :( 

Any suggestions on how to try get it some traction? Seems like a pretty straightforward thing to approve and merge

**Comment by artiomchi** [comment]
Any suggestions on how to get this moving from the team? There's been no movement at all since I opened the PR on the 2nd of May. (#16698)
